app/src/features/Batches/BatchCreate.tsx

```diff
@@ -87,20 +118,43 @@ const BatchCreate = (): JSX.Element => {
     setSelectedResourceIds(event.target.value as string[]);
   };
 
-  const handleBatchRun = async () => {
-    try {
-      await apiBatchCreate({
-        batchRequest: {
-          resources: selectedResourceIds,
-        },
-      }).unwrap();
-
-      setSelectedResourceIds([]);
-    } catch (e) {
-      setAlert(e.message as string);
-    }
+  const handleBatchRun = () => {
+    refetchMappings();
+    setIsBatchStarting(true);
   };
 
+  useEffect(() => {
```

Comment:
Why using an effect instead of doing things inside `handleBatchRun`?

Reply:
It won't work. I call `refetchMappings` in `handleBatchRun` and I can't get the updated data in the same scope. I need to wait for a rerender.

Reply:
It is a pity that the async `unwrap` method is not available for `refetch` imo

But if I got it right, this `useEffect` basically does what `handleBatchRun` would do if we could fetch the latest mappings directly in its scope